packages/ruleset/src/functions/use-date-based-format.js

```diff
@@ -0,0 +1,359 @@
+    const index = propertyPath.findIndex((prop, i) => prop !== examplePath[i]);
+    const value = getObjectValueAtPath(example, propertyPath.slice(index));
+
+    // If we find a value, go ahead and break from the loop.
+    if (value) {
+      logger.debug(
+        `${ruleId}: value found in example at location: ${examplePath.join(
+          '.'
+        )}`
+      );
+
+      exampleValue = value;
+      break;
+    }
+  }
+
+  logger.debug(
+    `${ruleId}: no example value found for schema at location: ${propertyPath.join(
+      '.'
+    )}`
+  );
+
+  // This will return `undefined` if we never find a value;
+  return exampleValue;
+}
+
+// This function takes an object, as well as a path to a specific value, and
+// recursively parses the object looking for the value at that path. If it
+// finds one, the value will be returned. If not, the function will return
+// `undefined`. One important note is that the array given as the `pathToValue`
+// argument *will* be modified by the logic, so if that is not desired, a copy
+// should be passed by the caller (using .slice(), for example).
+function getObjectValueAtPath(obj, pathToValue) {
+  // If obj is undefined, there is nothing to process.
+  if (obj === undefined) {
+    return;
+  }
+
+  // If we've exhausted the whole path, we've found the desired value!
+  if (!pathToValue.length) {
+    return obj;
+  }
+
+  const p = pathToValue.shift();
+
+  // Check for sentinel indicating an array.
+  if (p === '[]' && Array.isArray(obj) && obj.length) {
+    return getObjectValueAtPath(obj[0], pathToValue);
+  }
+
+  // Check for sentinel indicating a dictionary.
+  if (p === '{}' && isObject(obj) && Object.values(obj).length) {
+    return getObjectValueAtPath(Object.values(obj)[0], pathToValue);
+  }
+
+  // Standard model path.
+  if (obj[p]) {
+    return getObjectValueAtPath(obj[p], pathToValue);
+  }
+
+  // Return undefined if we don't find anything.
+  return;
+}
+
+// "Indirect" examples are those coming from request bodies, response bodies, and parameters.
+function checkForIndirectExamples(path, apidef) {
+  // Parameter and Media Type objects have the same format when it comes
+  // to examples, so we can treat all of these scenarios the same way.
+  if (
+    isRequestBodySchema(path) ||
+    isResponseSchema(path) ||
+    isParamSchema(path) ||
+    isParamContentSchema(path)
+  ) {
+    // Example fields would be siblings of the schema we're looking at, so we need to look in the API
+    // for the path, minus the last value (which is "schema").
+    const examples = getOpenApiExamples(
+      getObjectValueAtPath(apidef, path.slice(0, -1))
+    );
+
+    // Check for the special case of looking at a content schema for a parameter that
+    // itself defines an example (pull the last three values off the path to check).
+    if (isParamContentSchema(path)) {
+      examples.push(
+        ...getOpenApiExamples(getObjectValueAtPath(apidef, path.slice(0, -3)))
+      );
+    }
```

Comment:
I don't think you need to do this way down here — these sorts of examples only exist at the top level where the rule is called for each schema-usage location, right?

Reply:
That's right - this function is the first thing that's called when we look at a schema. If it's the top level, a schema in a schema-usage location, this will end up populating the list with any relevant examples. Other times, we'll call it but it will end up being skipped. I'm not sure I understand what you mean by "way down here" 🙂 is there another place we should check for these examples?

Reply:
My point was just that it's unnecessary and a bit wasteful to be checking if it's a request body, response, or parameter schema in the actual recursion.

Reply:
Oh I see - yes, that can be optimized. So, this feedback is more on the location of the _invocation_ of this function - I'll update that.